### PR TITLE
Enable CP Server docker images to work in KRaft mode

### PR DESCRIPTION
**Pull request by stejani-cflt**
This change enables the CP server docker images to work in KRaft mode.  This will be enabled in CP 7.4 where KRaft is GA for new clusters.

Changes required include:
- There is a new environment variable `KAFKA_CONTROLLER_QUORUM_MODE` that if not specified falls back to Zookeeper mode.  If KRaft mode it does things required to bring it up in KRaft mode
- In KRaft mode, the disk needs to be formatted with a unique cluster id that is specified in the `docker-compose.yml` file
- `KAFKA_ADVERTISED_LISTENERS` is not required when booting up a controller in KRaft mode. This requires it to be unset, because there isn't a way to conditionally add it in `dockerfile.ubi8`
- `KAFKA_ZOOKEEPER_CONNECT` and as a result Zookeeper health check not required in KRaft mode
- Example `docker-compose.yml` file

### CODE REVIEW

File: server/include/etc/confluent/docker/ensure [hunk 1]
```diff
@@ -23,7 +23,17 @@ dub path "$KAFKA_DATA_DIRS" writable
...
 else
     echo "===> Check if Zookeeper is healthy ..."
     cub zk-ready "$KAFKA_ZOOKEEPER_CONNECT" "${KAFKA_CUB_ZK_TIMEOUT:-40}"
 fi
+
+# KRaft required step: Format the storage directory with provided cluster ID
+if [[ -n "${KAFKA_CONTROLLER_QUORUM_MODE-}" ]] && [[ $KAFKA_CONTROLLER_QUORUM_MODE == "kraft" ]]
+then
+    echo "===> Using provided cluster id $KAFKA_CLUSTER_ID ..."
+    kafka-storage format --cluster-id=$KAFKA_CLUSTER_ID -c /etc/kafka/kafka.properties
```
Comment (stejani-cflt) [inline server/include/etc/confluent/docker/ensure:38]: Still need to edit this to not format it if it `meta.properties` already exists

**Comment by gracechensd** [comment]
Can we target 7.4.x branch? Then after this is merged, can pint merge it to master
> Do I need to do this for the kafka images too (I guess these are from CCS)?

Unsure about for kafka images, I'll let others in kafka-eng comment

**Comment by stejani-cflt** [comment]
Moving this PR to 7.4.x - https://github.com/confluentinc/kafka-images/pull/209

**Comment by forlack** [comment]
We should not introduce a new kafka configuration for Quorum mode.  This does not exist in AK.  We should detect if the user is setting KRaft properties like server.roles.  And use KRaft mode in that case

**Pull request closed without merging**